src/rrd_cgi.c

```diff
@@ -560,13 +559,13 @@ static char *rrdsetenv(
     const char **args)
 {
     if (argc >= 2) {
-        const size_t len = strlen(args[0]) + strlen(args[1]) + 2;
-        char *xyz = (char *) malloc(len);
+        char *xyz = NULL;
 
+        asprintf(&xyz, "%s=%s", args[0], args[1]);
```

Comment:
> we have been avoiding asprintf since it is a GNU extension ...

noo, we've been using it and providing a shim, which is the right thing to do.  https://github.com/oetiker/rrdtool-1.x/search?q=asprintf

> The FreeBSD implementation sets strp to NULL on error.

Good for them, but we don't need to rely on that.  FreeBSD still return -1 on failure.